Question2.java

```diff
@@ -0,0 +1,137 @@
+//SinglyLinkedList class
+class SinglyLinkedList<Type>
+{
+    Node head;
+
+    private class Node //Create node
+    {
+        Type data;
+        Node next;
+    }
+
+    public void add_end(Type data) //Add to end of list
+    {
+        Node node = new Node();
+        node.data = data;
+        node.next = null;
+
+        if(head==null) head = node;
+
+        else
+        {
+            Node cursor = head;
+            while(cursor.next!=null)
+            {
+                cursor = cursor.next;
+            }
+            cursor.next = node;
+        }
+
+    }
+    public void add_start(Type data) //Add to start of list
+    {
+        Node node = new Node();
+        node.data = data;
+        node.next = head;
+        head = node;
+    }
+
+    public void add_after(int index,Type data) //Add after an element
+    {
+        Node node = new Node();
+        node.data = data;
+        node.next = null;
+
+        Node cursor = head;
+        for(int i=0;i<=index-1;i++)
+        {
+            cursor = cursor.next;
+        }
+        node.next = cursor.next;
+        cursor.next = node;
+    }
+
+    public void delete_start() //Delete from start of list
+    {
+        head = head.next;
+    }
+
+    public void delete_end() //Delete from end of list
+    {
+        Node cursor = head;
+        int size=0;
+        while(cursor.next!=null)
+        {
+            cursor = cursor.next;
+            size+=1;
+        }
+        delete_after(size-1);
+    }
+    public void delete_after(int index) //Delete after an index
+    {
+        Node cursor = head;
+        Node delete_index = null;
+        for(int i=0;i<=index-1;i++)
+        {
+            cursor = cursor.next;
+        }
+        delete_index = cursor.next;
+        cursor.next = delete_index.next;
+
+        delete_index = null;
+    }
+
+    public void show() //Print list elements
+    {
+        Node node = head;
+        try
+        {
+            while (node.next != null)
+            {
+                System.out.println(node.data);
+                node = node.next;
+            }
+            System.out.println(node.data);
+        }
+        catch(NullPointerException e)
```

Comment:
Instead of try catch do a check first and print **List is empty!** if there are no elements. Else print the list.